metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -8268,29 +8279,90 @@ public NotificationEventResponse getNextNotification(NotificationEventRequest rq
     }
   }
 
+  private void lockForUpdate() throws MetaException {
+    String selectQuery = "select NEXT_EVENT_ID from NOTIFICATION_SEQUENCE";
+    String selectForUpdateQuery = sqlGenerator.addForUpdateClause(selectQuery);
```

Comment:
We found in our experiments that using REPEATABLE READ isolation a query similar to "select NEXT_EVENT_ID from NOTIFICATION_SEQUENCE where NNI_ID = 1" will also lock the row for update and block any transaction that attempts to update the same NNI_ID =1 row.

Do you think that would be better to use? The syntax will work on any DB. We did that test on Mysql, though.

Reply:
answered the same question from Alex on the jira.